automation_tools/scripts/ga-migration/main.py

```diff
@@ -0,0 +1,238 @@
+
+    content2 = contents.replace(m.group(0), py_parsed_string)
+
+    with open(filepath, "w") as f:
+        f.write(content2)
+
+
+@click.command()
+@click.option("--targetpath", default=".", help="Target repo directory path")
+def pipeline(targetpath):
+    """Helps the migration from Travis CI pipelines
+    to GitHub Actions running some common tasks"""
+
+    repo = targetpath.split("/")[-1]
+    repo_underscores = repo.replace("-", "_")
+
+    # TODO: add the trailing slash only if needed
+    targetpath = targetpath + "/"
+    # Reference: https://codimd.web.cern.ch/TOOkF5yhSAKJq3TiY0L42A?view
+
+    # .editorconfig
+    replace_simple(
+        targetpath + ".travis.yml", ".github/workflows/*.yml", ".editorconfig"
+    )
+
+    # README.rst
+    replace_regex(
+        r"https:\/\/img\.shields\.io\/travis\/([a-z]*\/[a-z-]*)\.svg",
+        "https://github.com/\\1/workflows/CI/badge.svg",
+        targetpath + "README.rst",
+    )
+    replace_regex(
+        r"https:\/\/travis-ci\.org\/([a-z]*\/[a-z-]*)",
+        "https://github.com/\\1/actions?query=workflow%3ACI",
+        targetpath + "README.rst",
+    )
+
+    # CONTRIBUTING.rst
+    replace_regex(
+        r"https:\/\/travis-ci\.(org|com)\/([a-z]*\/[a-z-]*)\/pull_requests",
+        "https://github.com/\\2/actions?query=event%3Apull_request",
+        targetpath + "CONTRIBUTING.rst",
+    )
+
+    # run-tests.sh
+    delete_line("isort", targetpath + "run-tests.sh")
+    replace_simple(
+        'check-manifest --ignore ".travis-*"',
+        'check-manifest --ignore ".*-requirements.txt"',
+        targetpath + "run-tests.sh",
+    )
+
+    # Download tests.yml template
+    download_file(
+        GA_TESTS_YAML_URL,
+        targetpath + ".github/workflows/tests.yml",
+    )
+
+    # Download pypi-publish.yml template
+    download_file(
+        GA_PYPI_PUBLISH_YAML_URL,
+        targetpath + ".github/workflows/pypi-publish.yml",
+    )
+
+    # pytest.ini
+    delete_line("pep8ignore", targetpath + "pytest.ini")
+    replace_regex(
+        "(addopts =).*",
+        f'\\1 --isort --pydocstyle --pycodestyle --doctest-glob="*.rst" --doctest-modules --cov={repo_underscores} --cov-report=term-missing',
+        targetpath + "pytest.ini",
+    )
+    if not file_contains("testpaths", targetpath + "pytest.ini"):
+        append_to_file(
+            f"testpaths = tests {repo_underscores}", targetpath + "pytest.ini"
+        )
+
+    # Add .github/workflows *.yml to MANIFEST.in
+    add_line(
+        "recursive-include .github/workflows *.yml", targetpath + "MANIFEST.in"
+    )
+
+    # Delete travis file
+    delete_file(targetpath + ".travis.yml")
+
+    # Remove bak files
+    delete_file(targetpath + "*.bak")
+
+    # replace_list(targetpath + "setup.py", ["a"], ["b"])
+    replace_list(
+        targetpath + "setup.py",
+        r"tests_require = ([[\s*\"(a-z-Z><=0-9.),]*])",
+        ["pytest-cov", "pytest-pep8"],
```

Comment:
Elements to be removed should be all the ones that [`pytest-invenio` installs](https://github.com/inveniosoftware/pytest-invenio/blob/798eaa4140dcdfe6e2cb51338915701a993545ee/setup.py#L48), plus probably `isort`, `pydocstyle`, etc. as these are now installed as `pytest-*` as you can see in `pytest-invenio`.

Reply:
https://github.com/inveniosoftware/automation-tools/pull/2/commits/ca6cb9fe1e40061fef5b2cdee3ff03f626026549